Fertige Klasse zur Punkte Berechnung

FishMacc opened the pull request:
Übergabe Paramater sind gesamtes Wort,Anfangs x Position, Anfangs y Position und ob horizonatal als boolean. Berechnung aller Werte wird in "Spiel" Klasse gemacht.

Commit:
Fertige Klasse zur Punkte Berechnung
Übergabe Paramater sind gesamtes Wort,Anfangs x Position, Anfangs y Position und ob horizonatal als boolean. Berechnung aller Werte wird in "Spiel" Klasse gemacht.

kiltz commented on ns-gui/src/main/java/de/nocompany/noscrabble/model/punktemultiplikator/punktemultiplikator.java:
```diff
@@ -0,0 +1,104 @@
+import java.util.HashMap;
+import java.util.Map;
+
+public class punktemultiplikator {
```
Namenskonventionen!

kiltz commented on ns-gui/src/main/java/de/nocompany/noscrabble/model/punktemultiplikator/punktemultiplikator.java:
```diff
@@ -0,0 +1,104 @@
+
+        return buchstabenPunkte;
+    }
+    public  int calcWortPunkte(String wort, int startx, int starty, boolean horizontal) {
```
Wo ist der Test dazu?

kiltz commented on ns-gui/src/main/java/de/nocompany/noscrabble/model/punktemultiplikator/punktemultiplikator.java:
```diff
@@ -0,0 +1,104 @@
+    }
+
+
+    public  int calcGesPunkte(int x, int y, Character letter) {
```
Tests!

kiltz commented on ns-gui/src/main/java/de/nocompany/noscrabble/model/punktemultiplikator/punktemultiplikator.java:
```diff
@@ -0,0 +1,104 @@
+    private static final int TW = 30; // Dreifacher Wortwert
+    private static final int DL = 2; // Doppelter Buchstabenwert
+    private static final int TL = 3; // Dreifacher Buchstabenwert
+    private static final int[][] punkteMultiplikatoren = {
```
Schaut sehr umständlich aus. Sollte perspektivisch in eine ENUM umgewandelt werden.

Commit:
Änderungen das Test funktionieren

Commit:
Multiplikator Enum Klasse

Commit:
Konstanten rausgenommen und durch ENUM getauscht
calcMultiplikator duch Punktemultiplikator unnötig
calcGesPUnkte durch Punktemultiplikator unnötig

Commit:
Tests für testCalcLetterPunkte und testCalcWortPunkte
